src/components/data/Table.tsx

```diff
@@ -6,17 +6,14 @@ import { useSyncedRef } from '../../hooks/use-synced-ref';
 import type { PresentationalProps } from '../../types';
 import { downcastRef } from '../../util/typing';
 import TableContext from './TableContext';
+import type { CommonTableInfo } from './TableContext';
 import type { TableInfo } from './TableContext';
 
-export type TableProps = PresentationalProps & {
-  stickyHeader?: boolean;
-  /** Sets accessible aria-label */
-  title: string;
-  /** This table has rows that can be selected */
-  interactive?: boolean;
-  /** Turn off outer table borders */
-  borderless?: boolean;
-} & Omit<JSX.HTMLAttributes<HTMLElement>, 'rows'>;
+export type TableProps = PresentationalProps &
+  CommonTableInfo & {
```

Comment:
`CommonTableInfo` is an internal type that wasn't intended to appear in the public API and isn't exported. I actually think the previous approach of duplicating the props was OK. If you did need to share the props, it would be better to do in the other way around with the internal type borrowing from the public API.